Spurning 1   þráinn

thrainn25 opened the pull request:
Endilega farið yfir kóðan hjá mér.

Til að keyra kóðan hjá ykkur þurfið þið að opna sqlite3 í terminal og síðan muna að vista kóðann áður en þið keyrið hann í VScode.

Keyrið svo kóðann með skipuninni .read code\names.sql
Hér er ég með code\ þar sem kóðinn minn er geymdur í möppunni code í VScode. Ef þið vistið kóðann á öðrum stað þurfið þið að breyta code í heitið á möppunni í VScode þar sem ykkar kóði er. Líka ef þið vistið þetta bara beint inn án þess að hafa möppu þá sleppið þið algjörlega þessu og hafið aðeins .read names.sql.

Tilgangur spurninga og kóða.
Ég hef í kóðanum í names.sql lýsingar fyrir hvern part í kóðanum hvað hann gerir. Eftir að kóðinn hefur lesið inn og búið til töflu fyrir upplýsingarnar úr bæði first_names_freq.csv og middle_names_freq.csv þá býr kóðinn til output file (úttak) sem sýnir og geymir niðurstöðurnar úr kóðanum.

Spurning eitt finnur út hvaða hópmeðlimur af okkur þrem hefur algengasta fyrsta nafnið og svarar því sem og hversu oft.
Spurning tvö finnur út hvaða nöfn frá okkur þrem sem sagt öll þrjú fyrstu nöfnin og tvö millinöfn eru alngengust og svarar ártali.
Spurning þrjú finnur út hvenær nöfnin fimm koma fyrst fram og svarar ártali.

thrainn25 committed:
Spurning 1

thrainn25 committed:
Merge branch 'Upplysingaverkfraedi:main' into Spurning-1---Þráinn

gigjamarin reviewed (changes requested):
Forritið mjög flott og keyrist alveg rétt. En mættir bara setja inn betri útskýringar við kóðunum. Ég á smá erfitt með að skilja hvað er í gangi. Kannski á íslensku líka :)

asdisgudmunds commented:
Þetta kemur í output skjalið hja mer: 
"Spurning 1: Hvaða hópmeðlimur á algengasta eiginnafnið??"
"Ásdís",24

En skil ekki hvað þetta 24 er hérna því það er ekki fjöldinn á Ásdísum

thrainn25 committed:
Breytingar á spurningu 1

Sjá hér breytingar og lagfæringu á spurningu 1. Lagaði bæði kóðann fyrir fyrstu spurninguna og setti betri lýsingu fyrir allar þrjár spurningarnar.

thrainn25 commented:
> Þetta kemur í output skjalið hja mer: "Spurning 1: Hvaða hópmeðlimur á algengasta eiginnafnið??" "Ásdís",24
> 
> En skil ekki hvað þetta 24 er hérna því það er ekki fjöldinn á Ásdísum

Þetta 24 var fjöldinn af línum sem Ásdís kemur giska ég. Ég lagaði kóðann þannig að summa allra count fyrir öll nöfnin. Ásdís var ennþá vinsælast og var niðurstaðan 1021 sinnum.

thrainn25 commented:
> Forritið mjög flott og keyrist alveg rétt. En mættir bara setja inn betri útskýringar við kóðunum. Ég á smá erfitt með að skilja hvað er í gangi. Kannski á íslensku líka :)

Ég var að setja inn annað skjal, með betri lýsingu sem og lagaði spurning 1.

asdisgudmunds reviewed (approved):
Flott

gigjamarin reviewed (approved):
Mjög flott, og góðar útskýringar :)
